internal/pkg/instrumentation/probe/probe.go

```diff
@@ -42,7 +42,7 @@ type Probe interface {
 	Load(*link.Executable, *process.TargetDetails, *sampling.Config) error
 
 	// Run runs the events processing loop.
-	Run(tracesChan chan<- ptrace.ScopeSpans)
+	Run(func(ptrace.ScopeSpans))
```

Comment:
Should we change it to?
```suggestion
	Run(func(ptrace.ScopeSpans) error)
```
Currently there is no error that I see that can be returned from that function, but I'm not sure.

Reply:
I was actually thinking of going the opposite direction: https://github.com/open-telemetry/opentelemetry-go-instrumentation/issues/1105#issuecomment-2458232414

The reason being that the controller and manager are higher up in the processing chain and already handle errors by logging or retrying. Having an error from these types returned to a low-level probe, especially one provided by a user, will create more of question about how they are expected to resolve the error.